### ISSUE TITLE
Implement notifications infrastructure layer for unified dashboard

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#4152]
As part of the unified dashboard epic, notifications will be rendered in a new container connected to the header bar (see #4050). This issue is about introducing the necessary infrastructure for these notifications, which will be separate from all existing dashboard notifications infrastructure, which is still legacy and using JS filters, which we are working our way away from.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* A new `assets/js/components/notifications/BannerNotifications` should be introduced, essentially as the container that controls rendering of any banner notifications in the unified dashboard (see #4153 on why this should be called `BannerNotifications`).
* The component should mostly take over/copy the logic from `assets/js/components/legacy-notifications/index.js`, which controls the current ("non-unified") dashboard.
    * It shouldn't use filters etc. of course, the main purpose is just to render the same notification components under the same circumstances.
    * One slight tweak should be that, _if_ the `DashboardSetupAlerts` component needs to be rendered, it should be the only component, i.e. just return that component. Also, the `'authentication_failure' === notification` condition can be omitted in the new component, since that value isn't actually used anywhere anymore. That `'authentication_failure'` condition should also be entirely removed from the `assets/js/components/legacy-notifications/dashboard-setup-alerts.js` component.
    * The only other tweak should be that the `DashboardModulesAlerts` component should be ignored, i.e. that condition shouldn't be included in the new component. That component still needs to be refactored and can be integrated into the new `BannerNotifications` component later.

## Implementation Brief
* Create `assets/js/components/notifications/BannerNotifications` which exports the `BannerNotifications` functional component.
* `BannerNotifications` should take most of the code/logic from `assets/js/components/legacy-notifications/index.js` with the following changes:
	* If `DashboardSetupAlerts` needs to be rendered, return the component only, i.e do not continue with rendering other components.
	* remove the use of filters to instead render the underlying components directly using the same conditionals.
	* Remove the `'authentication_failure' === notification` condition to display `DashboardSetupAlerts`. The same condition should be removed from `assets/js/components/legacy-notifications/dashboard-setup-alerts.js`.
	* Remove the use of `DashboardModulesAlerts` along with the corresponding condition.

### Test Coverage
* Failing tests should be updated to cater for above changes.

## QA Brief

* No user facing changes yet, will be QA'd on related #4168 and #4050. For now: verify the implemented code in #4264 satisfies the ACs.

## Changelog entry

* Update notifications for Unified Dashboard.

**Comment by tofumatt** [comment]
IB ✅

**Comment by felixarntz** [comment]
In order to unblock this issue, I've removed the last point from the ACs and IB so that this can be worked on now and doesn't need to wait for #4050. The removed point is now included in the ACs for #4050, i.e. the blocking order is now reversed.

**Comment by felixarntz** [comment]
## Approval ❌ 

The implementation of `BannerNotifications` from #4264 does not match the current behavior of what's in `assets/js/components/legacy-notifications/index.js` as required per the ACs:
* The new component only ever returns one of the notification components.
* The current `index.js` implementation allows for multiple notification components to show at the same time.
* The ACs state that the "only show one" requirement applies **only** to the setup success component, not in general.

Due to the above, the current implementation would lead to issues because for example as long as the user input feature flag is enabled (see https://github.com/google/site-kit-wp/pull/4264/files#diff-d4cc4257c04fff7ffff6f00b79230b01597c93c0359369c59a4e64380f00b8fbR42), the Idea Hub prompt notification would **never** trigger (see https://github.com/google/site-kit-wp/pull/4264/files#diff-d4cc4257c04fff7ffff6f00b79230b01597c93c0359369c59a4e64380f00b8fbR46).

@kuasha420 @tofumatt @asvinb Can one of you create a follow-up PR for `main` please? For all notifications other than `DashboardSetupAlerts`, the new component should return a `Fragment` with whatever notifications apply in the current conditions (or `null` if none of them apply).

**Comment by kuasha420** [comment]
@felixarntz on it. Sorry for misunderstanding the intent from the AC.

**Comment by felixarntz** [comment]
@kuasha420 Thank you, and no worries at all!

**Comment by kuasha420** [comment]
@felixarntz submitted the follow up pr. Took a bit long to figure out component rendering order, but should be good. Please review and if everything looks right, it can go straight to approval as it will be QA'd with the linked issues.

cc @eugene-manuilov

**Comment by techanvil** [comment]
**QA:Eng** :heavy_check_mark: Verified the result of PRs #4264 and #4319 against the ACs.

**Comment by felixarntz** [comment]
Approval ✅  Thanks @kuasha420!
